connectors/byoc.py

```diff
@@ -268,14 +271,34 @@ def get_draft_filter(self, domain=DEFAULT_DOMAIN):
     def get_filter(self, filter_state="active", domain=DEFAULT_DOMAIN):
         return next(
             (
-                filter_[filter_state]
+                Filter(filter_[filter_state])
                 for filter_ in self.filtering
                 if filter_["domain"] == domain
             ),
             {},
         )
 
 
+class Filter(dict):
```

Comment:
there's a cleaner way to extend a dict by the way, see https://stackoverflow.com/a/3387975/15097096